src/parser/jobs/sam/modules/Sen.tsx

```diff
@@ -1,98 +1,307 @@
 
 import Kenki from './Kenki'
 
-enum SEN {
-	SETSU = 'Setsu',
-	GETSU = 'Getsu',
-	KA = 'Ka',
-}
+const SEN_ACTIONS = [
+	ACTIONS.YUKIKAZE.id,
+	ACTIONS.GEKKO.id,
+	ACTIONS.MANGETSU.id,
+	ACTIONS.KASHA.id,
+	ACTIONS.OKA.id,
+]
 
-const SEN_ACTIONS = {
-	[ACTIONS.YUKIKAZE.id]: SEN.SETSU,
+// Setsu = Yuki, Getsu = Gekko Man, Ka = Kasha Oka
 
-	[ACTIONS.GEKKO.id]: SEN.GETSU,
-	[ACTIONS.MANGETSU.id]: SEN.GETSU,
+const SEN_REMOVERS = [
+	ACTIONS.HIGANBANA.id,
+	ACTIONS.TENKA_GOKEN.id,
+	ACTIONS.MIDARE_SETSUGEKKA.id,
+	ACTIONS.HAGAKURE.id,
+]
 
-	[ACTIONS.KASHA.id]: SEN.KA,
-	[ACTIONS.OKA.id]: SEN.KA,
-}
+const THINGS_WE_WANT_IN_THE_TABLE = [
+	ACTIONS.HAKAZE.id,
+	ACTIONS.JINPU.id,
+	ACTIONS.SHIFU.id,
+	ACTIONS.FUGA.id,
+
+	ACTIONS.GEKKO.id,
+	ACTIONS.KASHA.id,
+	ACTIONS.YUKIKAZE.id,
+	ACTIONS.MANGETSU.id,
+	ACTIONS.OKA.id,
 
-const IAIJUTSU = [
+	// Sen Spenders
 	ACTIONS.HIGANBANA.id,
 	ACTIONS.TENKA_GOKEN.id,
 	ACTIONS.MIDARE_SETSUGEKKA.id,
-]
+	ACTIONS.HAGAKURE.id,
 
-const TSUBAME = [
-	ACTIONS.KAESHI_HIGANBANA.id,
-	ACTIONS.KAESHI_GOKEN.id,
-	ACTIONS.KAESHI_SETSUGEKKA.id,
+	// I'm leaving these in as they are a way to handle Filler. not the usual way, but a way
+	ACTIONS.ENPI.id,
+	ACTIONS.HISSATSU_YATEN.id,
+	ACTIONS.HISSATSU_GYOTEN.id,
+
+	// OGCDS IGAF about.
+	ACTIONS.HISSATSU_KAITEN.id,
+	ACTIONS.MEIKYO_SHISUI.id,
 ]
 
 const KENKI_PER_SEN = 10
 
+const SEN_HANDLING = {
+	NONE: {priority: 0, message: 'No errors'},
+	OVERWROTE_SEN: {priority: 20, message: <Trans id = "sam.sen.sen_handling.overwrote_sen"> Contains a Overwrote Sen. </Trans>},
+	HAGAKURE: {priority: 10, message: <Trans id = "sam.sen.sen_handling.hagakure"> Contains a Hagakure. </Trans>},
+	DEATH: {priority: 30, message: <Trans id = "sam.sen.sen_handling.death"> Contains your death. </Trans>}, // BET YOU WISH YOU USED THIRD EYE NOW RED!
+}
+
+// God this grew outta control real fast
+
+class SenState {
+	start: number
+	end?: number
+	rotation: CastEvent[] = []
+	isNonStandard: boolean = false // Aka Hagakure + Overwrites, used to filter later.
+	isDone: boolean = false // I SWEAR TO GOD IF THIS JANK THING WORKS, I'M LEAVING IT
+	isDeath: boolean = false // DIE! DIE! DIE! -Reaper
+	isHaga: boolean = false // is it a haga or no?
+	isOverwrite: boolean = false // is it a overwrite or no?
+
+	// Sen State trackers, do I really need to explain?
+	currentSetsu: number = 0
+	currentGetsu: number = 0
+	currentKa: number = 0
+
+	overwriteSetsus: number = 0
+	overwriteGetsus: number = 0
+	overwriteKas: number = 0
+
+	kenkiGained: number = 0 // Kenki # * 10
+
+	_senCode: {priority: number, message: TODO} = SEN_HANDLING.NONE
+
+	public set senCode(code) {
+		if (code.priority > this._senCode.priority) {
+			this._senCode = code
+		}
+	}
+
+	public get senCode() {
+		return this._senCode
+	}
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
 export default class Sen extends Module {
-	static handle = 'sen'
+	static handle = 'Non-Standard Sen Windows'
```

Comment:
Handle should be a single camelCase “word”, in case other modules need to depend on this one. 

To change what displays in the sidebar, use the title static property and a trans tag or the t() macro -- e.g. 	
```suggestion
	static handle = 'sen'
	static title = t('sam.sen.title')`Non-Standard Sen Windows`
```

To use the t() macro, you will need to also `import {t} from '@lingui/macro'`